core/src/main/java/org/jclouds/config/ContextLinking.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.config;
+
+import org.jclouds.Context;
+import org.jclouds.View;
+
+import com.google.common.base.Function;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+import com.google.inject.TypeLiteral;
+import com.google.inject.name.Names;
+
+/**
+ * Utility methods to allow {@link Context} and {@link View} linking between
+ * contexts.
+ * <p>
+ * By using this module users can configure a context to be able to inject other
+ * contexts or views by their provider id.
+ */
+public class ContextLinking {
+
+   static final TypeLiteral<Supplier<Context>> CONTEXT_SUPPLIER = new TypeLiteral<Supplier<Context>>() {
+   };
+
+   static final TypeLiteral<Supplier<View>> VIEW_SUPPLIER = new TypeLiteral<Supplier<View>>() {
+   };
+
+   public static Module linkToView(final String id, final Supplier<View> view) {
```

Comment:
[minor] Perhaps call this `linkView` or `linkedView` rather than `linkToView`? If I read it, the idea that I'm "linking this view" and "linking that context" sounds a bit closer to what's happening than "I'm linking _to_ that view"?